docs/docs/Tutorials/ROS2/CPPTemplate.md

```diff
@@ -0,0 +1,394 @@
+```
+`...`
+``` cpp linenums="67" hl_lines="1"
+#endif  // YOUR_PACKAGE_NAME__CONTROL_POLICY_HPP_
+```
+
+Modify `CMakeLists.txt` as desired and add any dependencies in `package.xml` following standard
+ROS 2 documentation.
+
+---
+
+## Implement Controller
+Assuming you have followed the above,
+
+- `include/<your_package_name>/control_policy.hpp`
+- `src/controller.cpp`
+
+are stubbed out to implement your custom external controller. You may also use
+`config/controller.yaml` for any policy parameters.
+
+### ControlPolicy
+
+You may use the struct `ControlPolicy` in `control_policy.hpp` to implement your controller.
+
+``` cpp linenums="25"
+struct ControlPolicy
+{
+  // declare/init any parameter variables here
+  double foo{1.0};
+  double bar{10.0*foo};
+
+  ControlPolicy()
+  : foo{1.0},
+    bar{10.0*foo}
+  {
+    update_params();
+  }
+
+  // Update dependent variables after reading in params
+  void update_params()
+  {
+    bar = 10.0*foo;
+  }
+
+  // Modify function inputs as desired
+  // Calculate target value from feedback inputs
+  double target(
+    const double & /*some*/,
+    const double & /*feedback*/,
+    const double & /*values*/)
+  {
+
+    // secret sauce
+
+    return 0.0;  // obviously, modify to return proper target value
+  }
+};
+```
+
+- declare/define any configurable parameters in the struct and init list
+
+``` cpp linenums="27"
+  // declare/init any parameter variables here
+  double foo{1.0};
+  double bar{10.0*foo};
+
+  ControlPolicy()
+  : foo{1.0},
+    bar{10.0*foo}
+```
+
+- Set any dependent variables in `update_params` on line 39
+
+``` py linenums="38"
+  // Update dependent variables after reading in params
+  void update_params()
+  {
+    bar = 10.0*foo;
+  }
+```
+
+- Declare/get/update params in the `set_params` function of the `Controller` class on line 58
+
+``` cpp linenums="58"
+// Use ROS2 declare_parameter and get_parameter to set policy params
+void Controller::set_params()
+{
+  this->declare_parameter("foo", policy_->foo);
+  policy_->foo = this->get_parameter("foo").as_double();
+
+  // recompute any dependent variables
+  policy_->update_params();
+}
+```
+
+- Then, your control logic will go in the `target` function on line 46.
+    Modify the input args as well as the return value as necessary
+
+``` cpp linenums="44"
+  // Modify function inputs as desired
+  // Calculate target value from feedback inputs
+  double target(
+    const double & /*some*/,
+    const double & /*feedback*/,
+    const double & /*values*/)
+  {
+
+    // secret sauce
+
+    return 0.0;  // obviously, modify to return proper target value
+  }
+```
+
+### Controller
+
+The `Controller` class contains an instance of `ControlPolicy` as the member variable,
+`this->policy`. The `this->policy->target` function may be called anywhere within the
+`Controller` class. You may call it inside any of the data callbacks to enable feedback
+control (for example):
+
+``` cpp
```

Comment:
It's not mentioned which file this is in. Having a title bar would be helpful.
That wasn't a problem in the Python tutorial because there was just 1 file.

Having the file name in the paragraph above would still be helpful, because the `main()` block below is also in the same file, but maybe you don't want a title bar there since it isn't directly copy-pasted from the file. Your preference here.
```suggestion
``` cpp title="src/controller.cpp"
```

Reply:
This is just an example and isn't actual code in the file

Reply:
It would still be good to mention in the paragraph above, which file it intends for people to look at, since it references specific variable names, and the `power_callback` is a defined function. I was looking around for a file name, since that had been clear throughout the tutorial up to this point.